src/setupTests.ts

```
@@ -1,5 +1,58 @@
+// Mock Supabase client and other utilities globally using vi.mock for Vitest
+// Ensure these paths are correct relative to your project structure
+vi.mock('@/integrations/supabase/client', () => import('@/integrations/supabase/__mocks__/client'));
+vi.mock('@/lib/geminiUtils', () => import('@/lib/__mocks__/geminiUtils'));
+
+// Mock ResizeObserver for Radix UI components in JSDOM
+const MockResizeObserver = vi.fn(() => ({
+  observe: vi.fn(),
+  unobserve: vi.fn(),
+  disconnect: vi.fn(),
+}));
+// @ts-ignore
```

ESLint

[error] Use "@ts-expect-error" instead of "@ts-ignore", as "@ts-ignore" will do nothing if the following line is error-free.

(@typescript-eslint/ban-ts-comment)